subsys/bluetooth/host/adv.c

```diff
@@ -392,6 +391,119 @@ static int hci_set_ad_ext(struct bt_le_ext_adv *adv, uint16_t hci_op,
 	return bt_hci_cmd_send_sync(hci_op, buf, NULL);
 }
 
+static int hci_set_adv_ext_fragmented(struct bt_le_ext_adv *adv, uint16_t hci_op,
+				      const struct bt_ad *ad, size_t ad_len)
+{
+	struct bt_hci_cp_le_set_ext_adv_data *set_data;
+	struct net_buf *buf;
+	int err;
```

Comment:
The scope of these variables can be smaller if moved to the loops.

Reply:
Done